packages/react-instantsearch-core-next/src/useConnector.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+
+import { useIndexContext } from './useIndexContext';
+import { useStableValue } from './useStableValue';
+
+import type {
+  Connector,
+  WidgetDescription,
+  WidgetRenderState,
+} from 'instantsearch.js';
+
+type ConnectorState<TProps> = Omit<
+  WidgetRenderState<any, TProps>,
+  'widgetParams'
+>;
+
+export function useConnector<
+  TProps extends Record<string, unknown>,
+  TState extends ConnectorState<TProps>,
```

Comment:
isn't render state already part of the widget description? 

```ts
TDescription['renderState']
```

Reply:
Indeed, thanks! See 32c4ed8fcf12a54045043ab233eb849243659503.